fennel/datasets/datasets.py

```diff
@@ -241,11 +245,50 @@ def transform(self, func: Callable, schema: Dict = {}) -> _Node:
             return Transform(self, func, None)
         return Transform(self, func, copy.deepcopy(schema))
 
-    def filter(self, func: Callable) -> _Node:
+    def filter(self, func: Callable | Expr) -> _Node:
         return Filter(self, func)
 
-    def assign(self, name: str, dtype: Type, func: Callable) -> _Node:
-        return Assign(self, name, dtype, func)
+    def assign(self, *args, **kwargs) -> _Node:
+        """
+        Assign is an overloaded operator that can be used in several ways:
+        1. Assigning a new column with a lambda function:
+            >>> ds.assign("new_column", int, lambda x: x.old_column + 1)
+        2. Using kwarg syntax to assign a column:
+            >>> ds.assign(new_column=F("old_column") + 1).astype(int))
+        3. Assigning multiple columns:
+            >>> ds.assign(
+            ...     new_column1=(F("old_column1") + 1).astype(int),
+            ...     new_column2=(F("old_column2") + 2).astype(int),
+            ... )
+
+        """
+        # Check if the user is using the first syntax
+        # Skip args[1] because type check can be flaky
+        
+
+        # Skip args[1] check because type check can be flaky
+        # and args[0] and args[2] are always strings and callable 
+        if (
+            len(args) == 3
+            and isinstance(args[0], str)
+            and isinstance(args[2], Callable)
+        ):
+            return Assign(self, args[0], args[1], args[2])
+       
+        if len(kwargs) == 3:
+            # Check if the kwargs are name, dtype and func
+            if all(k in ["name", "dtype", "func"] for k in kwargs.keys()):
+                return Assign(
+                    self, kwargs["name"], kwargs["dtype"], kwargs["func"]
+                )
```

Comment:
Does this mean that when these magic kwargs are used, we can still fall to Pandas approach? Since this behavior is neither documented nor very ergonomic, I wonder if we should just remove it?